### PR TITLE
Nova flavor

### PR DESCRIPTION
**Pull request by hkumarmk**
Here are the dependent PRs.

JioCloud/puppet-nova#1
JioCloud/puppet-openstack_extras#2
JioCloud/puppet-openstacklib#1

### CODE REVIEW

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
oh stmon2 was not up :( testing again

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last time it failed because ct1 failed to connect to proxy for puppet installation

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by bodepd** [comment]
I'm going to assume this is the patch you want reviewed and close your other pull requests. Please re-open them if you also need them merged.

**Comment by bodepd** [comment]
please test this

File: hiera/data/common.yaml [hunk 1]
```diff
@@ -303,6 +307,29 @@ nova::compute::neutron::libvirt_vif_driver: nova_contrail_vif.contrailvif.VRoute
 rjil::nova::compute::cinder_rbd_secret_uuid: "%{hiera('cinder_rbd_secret_uuid')}"
 rjil::nova::compute::ceph_mon_key: "%{hiera('rjil::ceph::mon::key')}"
 
+rjil::nova::controller::nova_auth:
+  username: nova
+  password: "%{hiera('nova::keystone::auth::password')}"
```
Comment (bodepd) [inline hiera/data/common.yaml]: this exposes the password in the catalog

**Comment by bodepd** [comment]
replaced by #456

**Pull request closed without merging**